### PR TITLE
Rchain-2307/runtime manager/block approver protocol f in validate candidate

### PR DESCRIPTION
**Pull request by dzajkowski**
## Overview
<sup>_What this PR does, and why it's needed_</sup>
Yet another spot with cheap `F`.

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>
Rchain 2307

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: casper/src/main/scala/coop/rchain/casper/genesis/Genesis.scala [hunk 1]
```diff
@@ -72,27 +72,29 @@ object Genesis {
       initial: BlockMessage,
       startHash: StateHash,
       runtimeManager: RuntimeManager[Task]
```
Comment (goral09) [inline casper/src/main/scala/coop/rchain/casper/genesis/Genesis.scala]: Why not change this to `F` as well and return `F[BlockMessage]` ?

**Comment by dzajkowski** [comment]
bors try

**Comment by dzajkowski** [comment]
@JosDenmark the biuld is passing. see bors try runs. the automatic test run failed on sbt failure.

File: casper/src/main/scala/coop/rchain/casper/util/comm/BlockApproverProtocol.scala [hunk 2]
```diff
@@ -110,7 +113,7 @@ object BlockApproverProtocol {
       minimumBond: Long,
       maximumBond: Long,
       faucet: Boolean
-  )(implicit scheduler: Scheduler): Either[String, Unit] = {
+  )(implicit scheduler: Scheduler): F[Either[String, Unit]] = {
```
[the remark this answers is not in the document]
Comment (dzajkowski) [reply casper/src/main/scala/coop/rchain/casper/util/comm/BlockApproverProtocol.scala]: the scheduler cannot be dropped as `Task` is still used.

**Comment by dzajkowski** [comment]
bors try

**Comment by dzajkowski** [comment]
bors try

**Comment by ArturGajowy** [comment]
Please rebase the PR to get rid of the merge commit. I've checked, it does so cleanly :)

File: casper/src/test/scala/coop/rchain/casper/helper/HashSetCasperTestNode.scala [hunk 3]
```diff
@@ -55,7 +55,8 @@ class HashSetCasperTestNode[F[_]](
     val blockDagDir: Path,
     val blockStoreDir: Path,
     blockProcessingLock: Semaphore[F],
-    shardId: String = "rchain"
+    shardId: String = "rchain",
+    createRuntime: (Path, Long) => (RuntimeManager[Task], () => Task[Unit])
```
Comment (ArturGajowy) [inline casper/src/test/scala/coop/rchain/casper/helper/HashSetCasperTestNode.scala]: Shouldn't this be `(Path, Long) => Resource[Task, RuntimeManager[Task]]`?

Comment (dzajkowski) [reply casper/src/test/scala/coop/rchain/casper/helper/HashSetCasperTestNode.scala]: compromise deployed

File: casper/src/main/scala/coop/rchain/casper/util/comm/BlockApproverProtocol.scala [hunk 4]
```diff
@@ -146,27 +149,35 @@ object BlockApproverProtocol {
...
-                            .computeBonds(postState.postStateHash)
-                            .runSyncUnsafe(Duration.Inf)
-                        ).toEither
-                          .leftMap(_.getMessage)
-      tuplespaceBondsMap = tuplespaceBonds.map { case Bond(validator, stake) => validator -> stake }.toMap
-      _ <- (tuplespaceBondsMap == bonds)
-            .either(())
-            .or("Tuplespace bonds don't match expected ones.")
-    } yield ()
+    (validate match {
```
[the remark this answers is not in the document]
Comment (dzajkowski) [reply casper/src/main/scala/coop/rchain/casper/util/comm/BlockApproverProtocol.scala]: nice one!

**Comment by dzajkowski** [comment]
@JosDenmark could you re-review?

File: casper/src/test/scala/coop/rchain/casper/util/rholang/Resources.scala [hunk 5]
```diff
@@ -13,6 +14,8 @@ object Resources {
   )(implicit scheduler: Scheduler): Resource[Task, RuntimeManager[Task]] =
     mkRuntime(prefix)
       .flatMap { runtime =>
-        Resource.pure(RuntimeManager.fromRuntime(runtime))
+        Resource.make(RuntimeManager.fromRuntime[Task](runtime))(
+          _ => Task.unit /* FIXME close the manager */
```
Comment (KentShikama) [inline casper/src/test/scala/coop/rchain/casper/util/rholang/Resources.scala]: I prefer TODO: but as long as you remember.

**Comment by JosDenmark** [comment]
Looks good. The campaign continues!

**Pull request merged**